pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1780,6 +1853,21 @@ func (r *ReconcileClusterDeployment) deleteStaleProvisions(provs []*hivev1.Clust
 	}
 }
 
+// deleteOldFailedProvisions deletes the failed provisions which are more than 7 days old
+func (r *ReconcileClusterDeployment) deleteOldFailedProvisions(provs []*hivev1.ClusterProvision, cdLog log.FieldLogger) {
+	cdLog.Infof("Deleting failed provisions which are more than 7 days old")
+	for _, provision := range provs {
+		failedCond := controllerutils.FindClusterProvisionCondition(provision.Status.Conditions, hivev1.ClusterProvisionFailedCondition)
```

Comment:
You do not need to use the condition. You can use `provision.Spec.Stage`.

Reply:
condition also gives me `last transition time` which I use to precisely calculate when exactly status changed to failed. I can also use `creation timestamp` for the provision. Should not make much difference when we are considering 7 days time frame.

Reply:
Ah, of course. Leave it as is.